client/console_conf.go

```diff
@@ -0,0 +1,44 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import "time"
+
+// ConsoleConfStartResponse is the response for starting the console-conf
+// routine.
+type ConsoleConfStartResponse struct {
+	ActiveAutoRefreshChanges []string `json:"active-auto-refreshes,omitempty"`
+	ActiveAutoRefreshSnaps   []string `json:"active-auto-refresh-snaps,omitempty"`
+}
+
+// ConsoleConfStart starts the console-conf start routine.
+func (client *Client) ConsoleConfStart() ([]string, []string, error) {
+	resp := &ConsoleConfStartResponse{}
+	// do the post with a short timeout so that if snapd is not available due to
+	// maintenance we will return very quickly so the caller can handle that
+	opts := &doOptions{
+		Timeout: 2 * time.Second,
+		// XXX: it should be valid to set Retry to something to mean don't do
+		//      retries
```

Comment:
proposals? the main reason not to do that is indeed that Retry > Timeout works and it would be a bit more code

Reply:
eh I don't really have any proposals, just a random musing, I can delete this TODO if you think we should just stick to using Retry > Timeout. If that's the case perhaps we add a doc-comment to that effect to Retry on the doOptions struct.

Reply:
I'll delete this TODO and just add a doc-comment on the doOptions struct